lib/offline/storage.js

```diff
@@ -298,6 +298,9 @@ shaka.offline.Storage.prototype.remove = function(content) {
         drmEngine = new shaka.media.DrmEngine(
             netEngine, onError, function() {}, function() {});
         drmEngine.configure(this.player_.getConfiguration().drm);
+        // if the license is not stored, we dont need to enlist drmEngine in removing the license
+        if (!this.config_.isPersistentLicense) return Promise.resolve();
```

Comment:
Current config should not affect the remove() operation.  If the content you're removing has persistent licenses, you should still call `drmEngine.init()` and do the rest.

In particular, if there is no persistent license stored, `sessionIds` will be empty and `drmEngine.removeSessions()` will do nothing.  So I think this check can be removed.